### PR TITLE
Add isMounted ref to suppress async state setter functions

### LINKED ISSUE #14: Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function
Hey interesting issue - I'm redirecting a user after a sound and it looks like potentially a useEffect within use-sound isn't returning a way to unmount. Thoughts?

```js
  const [play,] = useSound(beepSfx)
  const history = useHistory()

  const onBeep = useCallback(() => play())

  const onCreate = useCallback(() => {
    onBeep()
    history.push(ROUTES.MANAGE_BRANDS)
  })
```

Console errors:

```
index.js:1 Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
    in WelcomeSelectPage (created by Context.Consumer)
  | console.<computed> | @ | index.js:1
-- | -- | -- | --
  | r | @ | react_devtools_backend.js:6
  | printWarning | @ | react-dom.development.js:88
  | error | @ | react-dom.development.js:60
  | warnAboutUpdateOnUnmountedFiberInDEV | @ | react-dom.development.js:23192
  | scheduleUpdateOnFiber | @ | react-dom.development.js:21200
  | dispatchAction | @ | react-dom.development.js:15682
  | (anonymous) | @ | index.ts:105
  | (anonymous) | @ | howler.js:1856
  | setTimeout (async) |   |  
  | _emit | @ | howler.js:1855
  | _ended | @ | howler.js:1920
  | setTimeout (async) |   |  
  | playWebAudio | @ | howler.js:843
  | play | @ | howler.js:855
  | (anonymous) | @ | index.ts:103
  | (anonymous) | @ | select.js:37
````

Line 37 is onBeep declaration

### PR DESCRIPTION
**Pull request by curt-tophatter**
@joshwcomeau I think this is what you were getting at. I'm having a hell of a time trying to test this locally though, due the following error:

Screen Shot 2020-11-21 at 1 41 45 PM

Fixes #14

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -1753,7 +1753,7 @@
   resolved "https://registry.yarnpkg.com/@types/history/-/history-4.7.5.tgz#527d20ef68571a4af02ed74350164e7a67544860"
   integrity sha512-wLD/Aq2VggCJXSjxEwrMafIP51Z+13H78nXIX0ABEuIGhmB5sNGbR113MOKo+yfw+RDo1ZU3DM6yfnnRF/+ouw==
 
-"@types/howler@^2.1.2":
+"@types/howler@2.1.2":
```
Comment (curt-tophatter) [inline yarn.lock:1756]: This just came from running a `yarn install` fwiw

File: .nvmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+13.12
```
Comment (curt-tophatter) [inline .nvmrc:1]: I had to upgrade my node version to get `yarn install` to pass as per https://github.com/nodejs/node/issues/32852#issuecomment-629692323

**Review by joshwcomeau — APPROVED** [review]
This looks good to me! After it's merged I'll give it a quick local test, and if everything looks good it'll be published as a patch.

Thanks!

**Comment by curt-tophatter** [comment]
🙌

**Comment by joshwcomeau** [comment]
Looks good! Published under 2.0.1.

RE: The hooks warning in your screenshot, I'm guessing that's the really-annoying issue with Yarn/NPM link. I've dealt with that in the past, but I forget how 😬 I think by linking React/ReactDOM as well? I test this project exclusively in the contained storybook.

Thanks again =)

**Comment by curt-tophatter** [comment]
Can't thank you enough for your time merging and publishing this!

**Comment by tillschomborg** [comment]
It is still there, isn't it?
See my comment and code here:
https://github.com/joshwcomeau/use-sound/issues/42#issuecomment-738619584

**Pull request merged**